### PR TITLE
Explore careers base service

### PR DESCRIPTION
**Pull request by syatessoftware**

### CODE REVIEW

File: DFC.App.ExploreCareers.Data/Contracts/IApiDataModel.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace DFC.App.ExploreCareers.Data.Contracts
+{
+    public interface IApiDataModel
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Data/Contracts/IApiDataModel.cs:5]: You do not  need to define this interface yourself as it is defined in the content nuget

File: DFC.App.ExploreCareers.Data/Contracts/IApiDataProcessorService.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace DFC.App.ExploreCareers.Data.Contracts
+{
+    public interface IApiDataProcessorService
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Data/Contracts/IApiDataProcessorService.cs:7]: You do not  need to define this interface yourself as it is defined in the content nuget

File: DFC.App.ExploreCareers.Data/Contracts/IApiService.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace DFC.App.ExploreCareers.Data.Contracts
+{
+    public interface IApiService
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Data/Contracts/IApiService.cs:7]: You do not  need to define this interface yourself as it is defined in the content nuget

File: DFC.App.ExploreCareers.Data/Contracts/IContentCacheService.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DFC.App.ExploreCareers.Data.Contracts
+{
+    public interface IContentCacheService
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Data/Contracts/IContentCacheService.cs:6]: You do not  need to define this interface yourself as it is defined in the content nuget

File: DFC.App.ExploreCareers.Data/Contracts/ICosmosRepository.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Net;
+using System.Threading.Tasks;
+using DFC.App.ExploreCareers.Data.Models;
+using DFC.Compui.Cosmos.Contracts;
+
+namespace DFC.App.ExploreCareers.Data.Contracts
+{
+    public interface ICosmosRepository<T>
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Data/Contracts/ICosmosRepository.cs:11]: You do not  need to define this interface yourself as it is defined in the Cosmos nuget

File: DFC.App.ExploreCareers.Repository.CosmosDb/CosmosDbConnection.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace DFC.App.ExploreCareers.Repository.CosmosDb
+{
+    [ExcludeFromCodeCoverage]
+    public class CosmosDbConnection
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Repository.CosmosDb/CosmosDbConnection.cs:7]: You do not  need to define this interface yourself as it is defined in the Cosmos nuget

File: DFC.App.ExploreCareers.Repository.CosmosDb/CosmosRepository.cs [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Threading.Tasks;
+using DFC.Compui.Cosmos.Contracts;
+
+namespace DFC.App.ExploreCareers.Repository.CosmosDb
+{
+    [ExcludeFromCodeCoverage]
+    public class CosmosRepository<T> : Data.Contracts.ICosmosRepository<T>
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers.Repository.CosmosDb/CosmosRepository.cs:19]: You do not  need to define this class yourself as it is defined in the Cosmos nuget

File: DFC.App.ExploreCareers/Startup.cs [hunk 8]
```diff
@@ -36,22 +48,82 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+            var cosmosDbConnectionContentPages = configuration.GetSection(CosmosDbContentPagesConfigAppSettings).Get<CosmosDbConnection>();
+            services.AddDocumentServices<JobCategory>(cosmosDbConnectionContentPages, env.IsDevelopment());
+
+            services.AddApplicationInsightsTelemetry();
+            services.AddHttpContextAccessor();
+            services.AddTransient<IEventMessageService<JobCategory>, EventMessageService<JobCategory>>();
+            services.AddAutoMapper(typeof(Startup).Assembly);
+            services.AddHostedServiceTelemetryWrapper();
+            services.AddSubscriptionBackgroundService(configuration);
+            services.AddTransient<IDocumentService<JobCategory>, DocumentService<JobCategory>>();
```
Comment (IanCrisp) [inline DFC.App.ExploreCareers/Startup.cs:79]: services.AddTransient<IDocumentService<JobCategory>, DocumentService<JobCategory>>();

Is not needed as this is done in the Comsos nuget as part of AddDocumentServices

**Review by IanCrisp — COMMENTED** [review]
A few Cosmos nuget related comments
Also, are you using the content nuget? a few comments about that too

**Review by IanCrisp — APPROVED** [review]

**Review by trivedihimanshu — APPROVED** [review]

**Pull request merged**